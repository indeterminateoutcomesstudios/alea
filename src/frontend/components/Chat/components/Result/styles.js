// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'

export const Container = styled.div`
  margin-top: 0.6rem;
  padding: 0;
  background: ${props => props.theme.backgroundSecondary};
  border-left: 3px solid ${colors.blue};
  padding-left: 5px;
  border-radius: 5px;
`

export const Row = styled.div`
  border-top: 1px solid ${props => props.theme.borderColor};

  &:first-child {
    border-top: 0;
  }
`

export const Modifier = styled.span`
  font-weight: 400;
  color: ${props => props.theme.colorSecondary};
  margin-right: 0.4em;
`

export const Roll = styled.span`
  font-weight: 600;
  margin-right: 0.4em;
`
