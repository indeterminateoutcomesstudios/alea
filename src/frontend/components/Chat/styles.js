// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  ${props =>
    props.isPinned
      ? `
    position: absolute;
    top: 4rem;
    right: 0;
    bottom: 0;
  `
      : `
    position: absolute;
    top: auto;
    right: 1rem;
    bottom: 1rem;
    box-shadow: 0 3px 5px 0 rgba(0, 0, 0, 0.2);
    border-radius: 5px;
  `};
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading

export const MessagesWrapper = styled.div`
  min-height: 0;
  flex: 1;
`
export const Messages = styled.div`
  overflow-y: ${props => (props.isPinned ? 'scroll' : 'hidden')};
  overflow-x: hidden;
  width: 315px;
  height: 100%;

  display: flex;
  flex-direction: column;
  flex: ${props => (props.isPinned ? '1' : 'none')};
`
