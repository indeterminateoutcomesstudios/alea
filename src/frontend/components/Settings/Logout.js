// @flow
import React from 'react'
import styled from 'styled-components'
import Button from '../Button'

type Props = {
  performLogout: Function
}

const Container = styled.div`
  margin-top: 2rem;
  display: flex;
`

const Logout = (props: Props) => {
  return (
    <Container>
      <Button red onClick={props.performLogout}>
        Logout
      </Button>
    </Container>
  )
}

export default Logout
