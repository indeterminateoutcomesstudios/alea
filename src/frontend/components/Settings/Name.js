// @flow
import React from 'react'
import Input from 'frontend/components/Input'

type Props = {
  name: string,
  saveDisplayName: Function
}

type State = {
  name: string
}

export default class Name extends React.Component<Props, State> {
  state = {
    name: this.props.name
  }

  componentWillReceiveProps(prev: Props) {
    // See if there's a new name
    if (this.props.name !== prev.name) {
      this.setState({ name: this.props.name })
    }
  }

  onChange = (e: SyntheticEvent<HTMLInputElement>) => {
    this.setState({ name: e.currentTarget.value })
  }

  onBlur = () => {
    if (this.state.name != this.props.name) {
      this.props.saveDisplayName(this.state.name)
    }
  }

  render() {
    return (
      <Input
        value={this.state.name}
        onChange={this.onChange}
        onBlur={this.onBlur}
      />
    )
  }
}
