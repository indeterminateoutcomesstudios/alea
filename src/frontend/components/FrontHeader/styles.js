// @flow
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  color: white;
  height: 8rem;
`

export const LogoButton = styled(Link)`
  padding-top: 27px;
  padding-left: 2rem;
  text-decoration: none;
  color: white;
`

export const Button = styled(Link)`
  background-color: white;
  border-radius: 5px;
  height: 4rem;
  line-height: 4rem;
  padding: 0 2rem;
  margin: 2rem;
  margin-left: auto;
  color: ${props => props.theme.backgroundInverted};
  font-family: ${fonts.heading};
  font-weight: 700;
  text-decoration: none;
`
