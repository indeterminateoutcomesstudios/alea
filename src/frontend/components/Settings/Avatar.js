// @flow
import * as React from 'react'
import AvatarEditor from 'react-avatar-editor'
import Button from 'frontend/components/Button'
import { FileInputWrapper, Current } from './styles'

type Props = {
  avatar: ?string,
  updateUserAvatar: (avatar: string) => void
}
type State = {
  isEditing: boolean,
  file: ?string,
  zoom: number
}

class Avatar extends React.Component<Props, State> {
  fileInput = React.createRef()
  imageScale = React.createRef()
  editor = React.createRef()

  state = {
    isEditing: false,
    file: null,
    zoom: 1
  }

  chooseFile = () => {
    const fileInput = this.fileInput.current
    if (!fileInput) {
      return
    }
    fileInput.click()
  }

  cancelEdits = () => {
    this.setState({
      isEditing: false,
      file: null,
      zoom: 1
    })
  }

  saveImage = () => {
    const editor = this.editor.current
    if (!editor) {
      return
    }
    const resizedCanvas = editor.getImageScaledToCanvas()
    const imageUrl = resizedCanvas.toDataURL()

    this.props.updateUserAvatar(imageUrl)

    this.setState({
      isEditing: false,
      file: null,
      zoom: 1
    })
  }

  imageSelected = () => {
    const fileInput = this.fileInput.current
    if (!fileInput) {
      return
    }
    this.setState({
      isEditing: true,
      file: fileInput.files[0]
    })
  }

  updateZoom = (e: SyntheticEvent<HTMLInputElement>) => {
    this.setState({
      zoom: Number(e.currentTarget.value)
    })
  }

  render() {
    const { avatar } = this.props
    const { isEditing, file, zoom } = this.state

    if (isEditing) {
      return (
        <div>
          <AvatarEditor
            ref={this.editor}
            image={file}
            width={200}
            height={200}
            border={20}
            color={[127, 127, 127, 0.6]}
            scale={zoom}
          />
          <div>
            <input
              type="range"
              min="1"
              max="6"
              step="0.1"
              value={zoom}
              onChange={this.updateZoom}
            />
          </div>
          <Button onClick={this.saveImage}>Save Picture</Button>{' '}
          <Button red onClick={this.cancelEdits}>
            Cancel
          </Button>
        </div>
      )
    }

    return (
      <div>
        <Current>
          {avatar ? (
            <img src={avatar} />
          ) : (
            "Looks like you don't have an avatar yet. Want to upload one?"
          )}
        </Current>
        <Button onClick={this.chooseFile}>Upload new picture</Button>
        <FileInputWrapper>
          <input
            type="file"
            ref={this.fileInput}
            accept="image/*"
            onInput={this.imageSelected}
            onClick={e => (e.target.value = null)}
          />
        </FileInputWrapper>
      </div>
    )
  }
}

export default Avatar
