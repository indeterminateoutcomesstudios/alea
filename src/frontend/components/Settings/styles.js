// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'

export const Row = styled.div`
  margin-bottom: 2rem;
`

export const ThemeButtons = styled.div`
  display: flex;
`

export const ThemeContainer = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  background-color: ${props => props.theme.map};
  border: 3px solid ${props => (props.selected ? colors.green : colors.gray)};
  width: 200px;
  height: 130px;
  margin-right: 2rem;
`

export const ThemeHeader = styled.div`
  background-color: ${props => props.theme.backgroundInverted};
  height: 15px;
  width: 100%;
`

export const ThemeArea = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
`

export const ThemeSidebar = styled.div`
  height: 100%;
  width: 50px;
  border-top: 15px solid ${props => props.theme.backgroundSecondary};
  background: ${props => props.theme.background};
`

export const ThemeChat = styled.div`
  width: 50px;
  height: 42px;
  border-radius: 4px;
  margin: auto 5px 5px auto;
  background-color: ${props => props.theme.background};
  border-top: 5px solid ${props => props.theme.backgroundInverted};
  border-bottom: 3px solid ${props => props.theme.backgroundSecondary};
`
