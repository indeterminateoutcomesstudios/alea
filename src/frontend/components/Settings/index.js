// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import Page from 'frontend/components/Page'
import Heading from 'frontend/components/Heading'
import Label from 'frontend/components/Label'
import Button from 'frontend/components/Button'
import Name from './Name'
import {
  Row,
  ThemeButtons,
  ThemeContainer,
  ThemeHeader,
  ThemeArea,
  ThemeSidebar,
  ThemeChat
} from './styles'
import * as themes from 'frontend/styles/themes'
import type { ThemeName } from 'common/types'
import type { Theme } from 'frontend/styles/themes'

type Props = {
  theme: Theme,
  performLogout: Function,
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  displayName: string
}

const Settings = (props: Props) => {
  const {
    displayName,
    changeDisplayName,
    theme,
    changeTheme,
    performLogout
  } = props

  const isLightTheme = theme.name == 'light'
  const isDarkTheme = theme.name == 'dark'

  return (
    <Page>
      <Heading>Profile</Heading>
      <Row>
        <Label>Display Name:</Label>
        <Name name={displayName} saveDisplayName={changeDisplayName} />
      </Row>

      <Heading>Preferences</Heading>
      <Row>
        <Label>Theme</Label>

        <ThemeButtons>
          <ThemeProvider theme={themes['light']}>
            <ThemeContainer
              selected={isLightTheme}
              onClick={() => changeTheme('light')}
            >
              <ThemeHeader />
              <ThemeArea>
                <ThemeSidebar />
                <ThemeChat />
              </ThemeArea>
            </ThemeContainer>
          </ThemeProvider>
          <ThemeProvider theme={themes['dark']}>
            <ThemeContainer
              selected={isDarkTheme}
              onClick={() => changeTheme('dark')}
            >
              <ThemeHeader />
              <ThemeArea>
                <ThemeSidebar />
                <ThemeChat />
              </ThemeArea>
            </ThemeContainer>
          </ThemeProvider>
        </ThemeButtons>
      </Row>
      <Row>
        <Label>Logout</Label>
        <Link to="/logout">
          <Button red onClick={performLogout}>
            Logout
          </Button>
        </Link>
      </Row>
    </Page>
  )
}

export default Settings
