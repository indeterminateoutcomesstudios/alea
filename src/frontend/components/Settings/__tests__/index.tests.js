// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { ThemeProvider } from 'styled-components'
import Settings from '../index.js'
import { light } from 'frontend/styles/themes'

describe('Settings component', () => {
  const renderer = new ShallowRenderer()

  it('renders correctly', () => {
    renderer.render(
      <ThemeProvider theme={light}>
        <Settings
          displayName="test"
          currentUserId="test"
          changeDisplayName={() => {}}
          theme={light}
          changeTheme={() => {}}
          performLogout={() => {}}
        />
      </ThemeProvider>
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
