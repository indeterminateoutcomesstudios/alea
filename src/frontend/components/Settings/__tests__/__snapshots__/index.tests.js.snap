// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<Settings
  changeDisplayName={[Function]}
  changeTheme={[Function]}
  currentUserId="test"
  displayName="test"
  performLogout={[Function]}
  theme={
    Object {
      "background": "#f7f7f7",
      "backgroundInverted": "#3c3c3c",
      "backgroundInvertedSecondary": "#555",
      "backgroundSecondary": "#ebe3d6",
      "borderColor": "#cdc5ba",
      "color": "black",
      "colorInverted": "#f7f7f7",
      "colorMedium": "#999",
      "colorSecondary": "#3c3c3c",
      "map": "#bfbca5",
      "modalBackground": "#f7f7f7",
      "modalOverlay": "rgba(0, 0, 0, 0.5)",
      "name": "light",
    }
  }
/>
`;
